CONTRIBUTING.rst

```diff
@@ -0,0 +1,201 @@
+
+
+Ground Rules
+************
+
+As a contributor, it is vital that we all follow a few conventions:
+
+-  Be welcoming to newcomers and encourage diverse new contributors from
+   all backgrounds. See the :ref:`code_of_conduct`.
+-  Create issues for changes and enhancements, this ensures that
+   everyone in the community has a chance to comment
+-  Use our ``.editorconfig`` file (see `Editor
+   Config <#editor-config>`__)
+-  Ensure that you pass all the tests before making a pull request
+-  Avoid pushing directly to master, all changes should come via pull
+   requests
+
+
+Setup
+*****
+
+Editor Config
+~~~~~~~~~~~~~
+
+The repository contains a ``.editorconfig`` file. This ensures that all
+of our text editors behave the same way and avoids spurious changes
+simply due to differing whitespace or end of line rules.
+
+Many editors have built in support for ``Editorconfig`` but some require
+a plugin. To work out if your editor requires a plugin, please check
+https://editorconfig.org/.
+
+
+Getting started
+***************
+
+Your First Contribution
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The development methodology for openscm makes heavy use of ``git``,
+``make``, virtual environments and test driven development. If you
+aren’t familiar with any of these terms it might be helpful to spend
+some time getting up to speed on these technologies. Some helpful
+resources (the longest take about 5 hours to work through):
+
+-  `git intro by software
+   carpentry <https://swcarpentry.github.io/git-novice/>`__
+-  `making a pull request <http://makeapullrequest.com/>`__
+-  `my first pull request <http://www.firsttimersonly.com/>`__
+-  `tests intro by software
+   carpentry <https://v4.software-carpentry.org/test/index.html>`__ and
+   `getting started with
+   mocking <https://semaphoreci.com/community/tutorials/getting-started-with-mocking-in-python>`__
+-  `make intro by software
+   carpentry <https://swcarpentry.github.io/make-novice/>`__
+-  `venv <https://docs.python.org/3/library/venv.html>`__
+
+
+Development workflow
+********************
+
+For almost all changes, there should be a corresponding `Pull
+Request <https://github.com/openclimatedata/openscm/pulls>`__ (PR) on
+GitHub to discuss the changes and track the overall implementation of
+the feature. These PRs should use the PR template.
+
+It is better to break a larger problem into smaller features if you can.
+Each feature is implemented as a branch and merged into master once all
+of the tests pass. This is instead of one long-lived branch which can be
+difficult to merge.
+
+The workflow for implementing a change to opencm is:
+
+-  Create a PR. Initially you will not be ready to merge so preface
+   the title of the PR with ‘WIP:’.
+-  Start a branch for the feature (or bug fix). When you start a new
+   branch, be sure to pull any changes to master first ``bash git
+   checkout master git pull git branch my-feature && git checkout
+   my-feature``
```

Comment:
split git instructions into multiple lines so it's clear where the new lines are